### DISCUSSION TITLE
Suggest to increase buffer size (`CURLOPT_BUFFERSIZE`) for multi downloads?

### DISCUSSION THREAD

**Discussion by jonashaag** [curl/curl#9611] [Ideas]
### I did this

In Mamba (see discussion here https://github.com/mamba-org/mamba/issues/1944) we use the multi interface to download Conda packages in parallel.

We found that downloads via Mamba were much slower than using `curl --parallel` and identified a different buffer size as the main cause. Here the buffer size is increased from 16 KiB to 100 KiB: https://github.com/curl/curl/blob/2481dbe5f4f581a3556b56cdc5a6d46be35e827d/src/tool_operate.c#L1290 When we do the same, throughput improves up to 10x.

### I expected the following

Personally I would've expected curl to have a better default buffer size for our use case but maybe that's not as easy. In any case I wonder if it makes sense to explicitly suggest increasing the buffer size in the multi interface docs.

### curl/libcurl version

libcurl 7.82.0

### operating system

Darwin Kernel Version 21.6.0: root:xnu-8020.140.49~2/RELEASE_ARM64_T8101 arm64

**Comment by bagder** [comment]
I don't think the size is particularly depending on what libcurl API you are using. If you are not resource constrained, it is probably a wise idea to bump it up to at least 100K no matter how you download.

I think the least we can do is to document the fact that increasing this size is a general good idea. The question is perhaps mostly *where* in the documentation this should be added to make developers notice?

A secondary question is of course if we should consider bumping the buffer size by default...

**Reply by xquery** [reply]
dumb question ... could (or would we even want to consider) buffer size be set dynamically by curl tool ... I think I know the answer but its a possibly third question ...

**Reply by bagder** [reply]
Sure, we *could* hypothetically set it dynamically with the tool, but I don't think we have yet found a good reason to add such an option. Also, it would be a option that is really hard for a user to understand when to use and what value to use for it.

The current default size set by the tool was done after measuring some speed differences with various sizes and going with one that seemed to be a good middle-ground. For localhost transfers, you might be able to squeeze out a little more transfer speed by increasing the size more.

**Reply by xquery** [reply]
sure ... was more thinking would be a nice 'knob' for testing purposes ...
